usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -197,6 +236,21 @@ def get_sort_by_fields(self):
 
         return sort_by_fields
 
+    def get_elastic_sort_by_fields(self):
+        if self.pagination["sort_key"] == "Award ID":
+            sort_by_fields = ["display_award_id"]
+        else:
+            if set(self.filters["award_type_codes"]) <= set(contract_type_mapping):
+                sort_by_fields = [contracts_mapping[self.pagination["sort_key"]]]
+            elif set(self.filters["award_type_codes"]) <= set(loan_type_mapping):
+                sort_by_fields = [loan_mapping[self.pagination["sort_key"]]]
+            elif set(self.filters["award_type_codes"]) <= set(idv_type_mapping):
+                sort_by_fields = [idv_mapping[self.pagination["sort_key"]]]
+            elif set(self.filters["award_type_codes"]) <= set(non_loan_assistance_type_mapping):
+                sort_by_fields = [non_loan_assist_mapping[self.pagination["sort_key"]]]
```

Comment:
```suggestion
                sort_by_fields = [non_loan_assist_mapping[self.pagination["sort_key"]]]
            else
                raise [some exception]
```
Since `self.filters` is provided by the user, it's quite possible that a request will qualify for none of the above cases, resulting in sort_by_fields being None and a 500 error later down the line. It would be better to explicitly raise a 400 error for an incorrect set of filters.

Reply:
The endpoint already checks for both empty filters and for incorrect award type codes and throws a 422, so it would be redundant.